src/zope/component/tests/test__api.py

```diff
@@ -1060,7 +1051,7 @@ def __conform__(self, iface):
             def queryUtility(self, iface, name, default):
                 if iface is IFactory and name == 'test':
                     return _factory
-                return default
+                raise NotImplementedError()
```

Comment:
Perhaps there should be some assertions (that iface is IFactory and name is test) instead. That way we capture the full intent of the test and break if anything changes.